linera-core/src/client/mod.rs

```diff
@@ -2469,60 +2356,131 @@ where
             }
         }
         self.update_from_info(&info);
-        let manager = &info.manager;
 
-        // If there is a validated block in the current round, finalize it.
-        if let Some(certificate) = &manager.requested_locked {
-            if certificate.round == manager.current_round {
+        let executed_block = if let Some(certificate) = &info.manager.requested_locked {
+            if certificate.round == info.manager.current_round {
+                // If there is a validated block in the current round, finalize it.
                 let committee = self.local_committee().await?;
                 match self.finalize_block(&committee, *certificate.clone()).await {
                     Ok(certificate) => return Ok(ClientOutcome::Committed(Some(certificate))),
                     Err(ChainClientError::CommunicationError(_)) => {
                         // Communication errors in this case often mean that someone else already
                         // finalized the block.
-                        let timestamp = manager.round_timeout.ok_or_else(|| {
+                        let timestamp = info.manager.round_timeout.ok_or_else(|| {
```

Comment:
Shouldn't we update our view of the chain (by querying for latest `chain_info_manager` again)?

Reply:
To be honest, I don't think that would be very useful; in fact, I don't think its strictly necessary to update the information in any case other than when we propose a block: If the chain height in the wallet file is not completely up to date, the client will synchronize next time it starts again anyway.